src/swsssdk/interface.py

```diff
@@ -149,9 +153,18 @@ def __init__(self, **kwargs):
         super(DBInterface, self).__init__()
 
         # Store the arguments for redis client
+        def _set(key):
+            value = os.environ.get(key[0])
+            if value:
+                kwargs[key[1]] = value
+
+        _set(self.REDIS_HOST)
+        _set(self.REDIS_PORT)
+        _set(self.REDIS_UNIX_SOCKET_PATH)
+
         self.redis_kwargs = kwargs
         if len(self.redis_kwargs) == 0:
-            self.redis_kwargs['unix_socket_path'] = self.REDIS_UNIX_SOCKET_PATH
```

Comment:
>redis_kwargs [](start = 17, length = 12)

We always prefer unix socket than net socket because of performance. That's the reason we fallback to default REDIS_UNIX_SOCKET_PATH if nothing provided in kwargs.